### PR TITLE
Add process group as arg to FSDP

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

Add process group as arg to FSDP

### CODE REVIEW

File: composer/trainer/mosaic_fsdp.py [hunk 1]
```diff
@@ -42,10 +42,21 @@ def patch_pytorch():
...
+        # Allow 2D HSDP
+        from torch.distributed.fsdp import _runtime_utils
+        _runtime_utils._validate_and_get_hybrid_shard_state = lambda *args, **kwargs: None
+
+    elif version.parse(torch.__version__) < version.parse('2.2.0'):
+        # Monkey path for torch < 2.2.0 ie torch == 2.1.1, 2.1.2
+
+        # Allow 2D HSDP
+        from torch.distributed.fsdp import _runtime_utils
+        _runtime_utils._validate_and_get_hybrid_shard_state = lambda *args, **kwargs: None
```
Comment (abhi-mosaic) [inline composer/trainer/mosaic_fsdp.py:62]: I see these monkey patches, but I assume our goal is to use `2.2.0` via torch nightly and not interact with any of these, is that right?

Just to double check, with `2.2.0`, do we not need any of these monkey patches?

Comment (mvpatel2000) [reply composer/trainer/mosaic_fsdp.py:62]: These are required for main torch. I have a PR open to torch to fix these in main but it has not been merged

**Review by vchiley — APPROVED** [review]

**Comment by vchiley** [comment]
for hsdp you set:
```
fsdp_config:
  process_group:
    - set<ReplicaSize>
    - mod<ReplicaSize>
```
?

**Comment by mvpatel2000** [comment]
> for hsdp you set:
> 
> ```
> fsdp_config:
>   process_group:
>     - set<ReplicaSize>
>     - mod<ReplicaSize>
> ```
> 
> ?

Yep!

**Pull request merged**